src/CommunityToolkit.Maui.MediaElement/Views/MediaManager.macios.cs

```diff
@@ -211,144 +213,110 @@ protected virtual partial void PlatformUpdateAspect()
 		};
 	}
 
-	protected virtual partial void PlatformUpdateSource()
+	protected virtual partial void PlatformUpdateSource() => Dispatcher.Dispatch(() =>
```

Comment:
Is the fixing a bug? 

We cannot call the dispatcher like this because it marshalls the code inside `Dispatcher.Dispatch()` to the main thread, causing `PlatformUpdateSource()` to return immediately before it has actually updated the source. 

If this is not fixing a bug, please remove this. If there is a bug, please link it in the PR description. 

If we need to marshall to the main thread here, we instead need to use `Dispatcher.DispatchAsync()`. This gets messy because `PlatformUpdateSource()` needs to `await` `DispatchAsync` and needs to be turned into an `async Task` method. Since this is a visual partial method, this means we then need to update each other class that uses `partial Task PlatformUpdateSource()`.

Reply:
I have no clue. I never wrote that! The only changes I have personally made was to loggers and a few minor tweaks. I am closing this PR and starting over. I just wanted to update some loggers. I will look at the suggested changes and reopen this again as only a logger pr. I am honestly super confused and did not make a bunch of these changes :(

Reply:
> Is the fixing a bug? 
> 
> We cannot call the dispatcher like this because it marshalls the code inside `Dispatcher.Dispatch()` to the main thread, causing `PlatformUpdateSource()` to return immediately before it has actually updated the source. 
> 
> If this is not fixing a bug, please remove this. If there is a bug, please link it in the PR description. 
> 
> If we need to marshall to the main thread here, we instead need to use `Dispatcher.DispatchAsync()`. This gets messy because `PlatformUpdateSource()` needs to `await` `DispatchAsync` and needs to be turned into an `async Task` method. Since this is a visual partial method, this means we then need to update each other class that uses `partial Task PlatformUpdateSource()`. 

I did not write this or add it to this PR.